pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -610,6 +613,25 @@ func (c *Controller) updateShootStatusOperationSuccess(ctx context.Context, g ga
 
 	return kutil.TryUpdateShootStatus(ctx, g, retry.DefaultRetry, shoot.ObjectMeta,
 		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+			if setConditionsToProgressing {
+				for i, cond := range shoot.Status.Conditions {
+					switch cond.Type {
+					case gardencorev1beta1.ShootAPIServerAvailable,
+						gardencorev1beta1.ShootControlPlaneHealthy,
+						gardencorev1beta1.ShootEveryNodeReady,
+						gardencorev1beta1.ShootSystemComponentsHealthy:
+						shoot.Status.Conditions[i].Status = gardencorev1beta1.ConditionProgressing
+					}
+				}
+				for i, constr := range shoot.Status.Constraints {
+					switch constr.Type {
+					case gardencorev1beta1.ShootHibernationPossible,
+						gardencorev1beta1.ShootMaintenancePreconditionsSatisfied:
+						shoot.Status.Constraints[i].Status = gardencorev1beta1.ConditionProgressing
+					}
+				}
```

Comment:
Should we also change reason/message? WDYT? I wouldn't change the times to keep the evaluation whether the status is `False` or still `Progressing` to the care controller and not duplicate it here, though.

Reply:
Hmm, also not too sure about this one. But I would propose to keep reason and message, set the status to `Progressing` and reset at least the `LastUpdateTime`.
This way, the thresholds for progressing conditions will count from the end of the reconciliation on. This makes sense, I guess, because we might change some stuff during reconciliation which might take a minute or so to be healthy again (e.g. loki/prometheus volumes).
WDYT?

Reply:
+1 for what @timebertt suggested

Reply:
The problem (or rather consequence) with this would be that a condition that was `False` already before/during the reconciliation would now become `Progressing` for some time. Maybe the reconciliation fixes it, however, we don't know and would show a "wrong" status until it transitions to `False` again.

Reply:
Hm, you have a point there. However, isn't that a problem anyway, i.e. we shortly set it to `Progressing` even now?
What if we only changed it to `Progressing` + `LastUpdateTime` update when the status was != `False` before?

Reply:
> i.e. we shortly set it to Progressing even now?

Hm, no, this is actually the purpose of `Progressing` and desired behaviour, right. But if something is already "broken" then we should probably not set it back to `Progressing`.

> What if we only changed it to Progressing + LastUpdateTime update when the status was != False before?

Nice, I like this. Will do!

Reply:
Agreed, sounds good 👍